### DISCUSSION TITLE
전역 변수의 네임스페이스 오염이란 무엇인가요?

### DISCUSSION THREAD

**Discussion by publdaze** [publdaze/js-study#57] [전역 변수의 문제점]
- 전역 변수의 네임스페이스 오염이란 무엇인가요?

**Comment by publdaze** [comment]
> - 전역 변수의 네임스페이스 오염이란 무엇인가요?
- 파일이 달라도 동일한 전역 스코프기 때문에 여러 파일에서 동일한 식별자가 존재할 때 원하지 않은 결과가 나올 수 있다는 걸 의미

**Comment by pipisebastian** [comment]
> 전역 변수의 네임스페이스 오염이란 무엇인가요?
 
- 파일이 분리되어있더라도, 하나의 전역 스코프를 공유하기 때문에, 각 파일에서 동일한 이름의 변수를 선언한다면, 값이 덮어씌게 되면서 의도치 않은 결과를 일으킵니다.
